pkg/core/data/ipc/transactions/provider.go

```diff
@@ -202,19 +206,19 @@ type provider struct {
 
 // GetBalance returns the balance of the client expressed in (unit?) of
 // DUSK. It accepts the ViewKey as parameter.
-func (p *provider) GetBalance(ctx context.Context, vk keys.ViewKey) (uint64, error) {
-	/*
-		rvk := new(rusk.ViewKey)
-		keys.MViewKey(rvk, &vk)
-		ctx, cancel := context.WithDeadline(ctx, time.Now().Add(p.timeout))
-		defer cancel()
-		res, err := p.client.GetBalance(ctx, &rusk.GetBalanceRequest{Vk: rvk})
-		if err != nil {
-			return 0, err
-		}
+func (p *provider) GetBalance(ctx context.Context, vk keys.ViewKey) (uint64, uint64, error) {
+	req := new(rusk.GetBalanceRequest)
+	req.Vk = new(rusk.ViewKey)
+	keys.MViewKey(req.Vk, &vk)
 
-	*/
-	return 0, nil
+	ctx, cancel := context.WithDeadline(ctx, time.Now().Add(p.txTimeout))
+	defer cancel()
+
+	res, err := p.walletClient.GetBalance(ctx, req)
+	if err != nil {
+		return 0, 0, err
+	}
```

Comment:
Tiny style nit: please leave a line between a closing curly brace and the next statement. I couldn't constrain this properly in the linter since it interfered with lots of other styling directives.